configure.ml

```diff
@@ -390,7 +390,7 @@ let coq_debug_flag = if !Prefs.debug then "-g" else ""
 let coq_profile_flag = if !Prefs.profile then "-p" else ""
 let coq_annotate_flag =
   if !Prefs.annotate
-  then if program_in_path "ocamlmerlin" then "-bin-annot" else "-annot"
+  then if program_in_path "ocamlmerlin" then "-bin-annot -annot" else "-annot"
```

Comment:
I would rather like this option to not depend on the programs that are in path at configure-time. Is there any problem with always using `-bin-annot` and `-annot`?

Reply:
I am not aware of any problems with using both annotations, except slight disk usage to store both.
Let me know if you want me to update the pull request to unconditionally generate both.

Reply:
`-annot` adds 160 MiB to a build, not something to be taken lightly IMHO.

Reply:
How about we just have both `-bin-annot` and `-annot` as `configure` flags? The user is already passing a flag, we can ask them what exactly they mean. If they want both they can pass both, etc.

Reply:
I like idea of 2 flags. The only problem is backward compatibility. It is logical not to generate any annotations in absence of any of 2 flags, but it such case it merlin will stop working for all developers until they re-configure with `-bin-annot` flag.

So my suggestion is:
1. Always generate `-bin-annot` regardless of merlin presence. There will be no way to disable this.
2. Optionally `-annot` could be specified to generate text annotations

Reply:
I don't think that backward compatibility is really an issue for developer-centric features.
I'm OK with both solutions: two configure flags, or one that generates both.

Reply:
I am new to the project and can't decide this. I can tweak the patch either way. I just want to be able to generate text annotations one way or another. If anybody else has an opinion one way or another, please comment.

Reply:
OK, well. If no one else has an opinion to share, I suggest going with @SkySkimmer's proposal. Reflecting the two possible options to `ocamlc` by having the exact same ones in `configure`. The `-annotate` option can be kept just to emit a warning suggesting to use either of these two options instead.

Reply:
I agree the simplest solution is to reflect the two flags at the `configure` level, and maybe keep the existing one with a deprecation warning.

Reply:
I've updated the pull requires to have 2 separate flags.